quark/utils.py

```diff
@@ -170,6 +170,31 @@ def str2bool(v):
     else:
         raise argparse.ArgumentTypeError('Boolean value expected.')
 
+def cmake_escape(s):
+    ln = len(s)
```

Comment:
Although not necessarily optimal, probably this would be easier to understand/validate and faster to execute:

```python
import re

def cmake_escape(s):
    # find longest equals sequence, add one equal
    equals = max(re.findall('=+', s), key=len, default='') + '='
    return '[%s[%s]%s]' % (equals, s, equals)
```